### PR TITLE
CFN fixes

### PR DESCRIPTION
**Pull request by arielkru**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Left to implement:
1. BaseResourceValueCheck for the cfn checks after RDSEncryption in the checks list
2. BaseResourceNegativeValueCheck class

### CODE REVIEW

File: checkov/cloudformation/checks/resource/base_resource_value_check.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+from abc import abstractmethod
+import dpath.util
+import re
+from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.cloudformation.context_parser import ContextParser
+from checkov.common.models.enums import CheckResult
+from checkov.common.models.consts import ANY_VALUE
+
+VARIABLE_DEPENDANT_REGEX = r'(?:local|var)\.[^\s]+'
```
Comment (schosterbarak) [inline checkov/cloudformation/checks/resource/base_resource_value_check.py:9]: cloudformation variables are different. i'll see how that can be implemented.

File: checkov/cloudformation/checks/resource/aws/DAXEncryption.py [hunk 2]
```diff
@@ -1,28 +1,21 @@
...
-            if conf['Properties'].get('SSESpecification'):
-                sse_specs_conf = conf['Properties']['SSESpecification']
-                if sse_specs_conf.get('SSEEnabled'):
-                    return CheckResult.PASSED
-        return CheckResult.FAILED
+    def get_inspected_key(self):
+        return 'Properties/SSESpecification/SSEEnabled'
+
+    def get_expected_value(self):
+        return ANY_VALUE
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/DAXEncryption.py:18]: @arielkru Should be `True`

File: checkov/cloudformation/checks/resource/aws/RDSPubliclyAccessible.py [hunk 3]
```diff
@@ -1,26 +1,21 @@
...
-from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
 
 
-class RDSPubliclyAccessible(BaseResourceCheck):
+class RDSPubliclyAccessible(BaseResourceValueCheck):
+    def get_expected_value(self):
+        return False
+
+    def get_inspected_key(self):
+        return 'Properties/PubliclyAccessible'
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/RDSPubliclyAccessible.py:10]: @arielkru Constructor should be first method in class

File: checkov/cloudformation/checks/resource/aws/cloudwatchLogGroupRetention.py [hunk 4]
```diff
@@ -1,26 +1,21 @@
...
-        """
+    def get_inspected_key(self):
+        return 'Properties/RetentionInDays'
 
-        if 'Properties' in conf.keys():
-            if 'RetentionInDays' in conf['Properties'].keys():
-                return CheckResult.PASSED
-        return CheckResult.FAILED
+    def get_expected_value(self):
+        return ANY_VALUE
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/cloudwatchLogGroupRetention.py:18]: @arielkru We've implemented it differently in terraform. Can we keep the consistency?

File: checkov/cloudformation/checks/resource/aws/S3Versioning.py [hunk 5]
```diff
@@ -1,20 +1,24 @@
+from checkov.cloudformation.checks.resource.base_resource_value_check import BaseResourceValueCheck
 from checkov.common.models.enums import CheckResult, CheckCategories
-from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
 
 
-class S3AccessLogs(BaseResourceCheck):
+class S3Versioning(BaseResourceValueCheck):
+    def get_inspected_key(self):
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/S3Versioning.py:6]: @arielkru Same constructor first comment

File: checkov/cloudformation/checks/resource/aws/S3RestrictPublicBuckets.py [hunk 6]
```diff
@@ -1,21 +1,18 @@
-from checkov.common.models.enums import CheckResult, CheckCategories
-from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.cloudformation.checks.resource.base_resource_value_check import BaseResourceValueCheck
+from checkov.common.models.enums import CheckCategories
 
 
-class S3RestrictPublicBuckets(BaseResourceCheck):
+class S3RestrictPublicBuckets(BaseResourceValueCheck):
+    def get_inspected_key(self):
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/S3RestrictPublicBuckets.py:6]: @arielkru Same constructor first comment

File: checkov/cloudformation/checks/resource/aws/S3PublicACLRead.py [hunk 7]
```diff
@@ -1,19 +1,21 @@
-from checkov.common.models.enums import CheckResult, CheckCategories
-from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.cloudformation.checks.resource.base_resource_negative_value_check import BaseResourceNegativeValueCheck
+from checkov.common.models.enums import CheckCategories
 
 
-class S3AccessLogs(BaseResourceCheck):
+class S3PublicACLRead(BaseResourceNegativeValueCheck):
+    def get_forbidden_values(self):
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/S3PublicACLRead.py:6]: @arielkru Same constructor first comment

File: checkov/cloudformation/checks/resource/aws/S3IgnorePublicACLs.py [hunk 8]
```diff
@@ -1,21 +1,17 @@
-from checkov.common.models.enums import CheckResult, CheckCategories
-from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.cloudformation.checks.resource.base_resource_value_check import BaseResourceValueCheck
+from checkov.common.models.enums import CheckCategories
 
 
-class S3IgnorePublicACLs(BaseResourceCheck):
+class S3IgnorePublicACLs(BaseResourceValueCheck):
+    def get_inspected_key(self):
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/S3IgnorePublicACLs.py:6]: @arielkru Same constructor first comment

File: checkov/cloudformation/checks/resource/aws/S3BlockPublicPolicy.py [hunk 9]
```diff
@@ -1,21 +1,17 @@
+from checkov.cloudformation.checks.resource.base_resource_value_check import BaseResourceValueCheck
 from checkov.common.models.enums import CheckResult, CheckCategories
 from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
 
 
-class S3BlockPublicPolicy(BaseResourceCheck):
+class S3BlockPublicPolicy(BaseResourceValueCheck):
+    def get_inspected_key(self):
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/S3BlockPublicPolicy.py:7]: @arielkru Same constructor first comment

File: checkov/cloudformation/checks/resource/aws/S3BlockPublicACLs.py [hunk 10]
```diff
@@ -1,21 +1,18 @@
+from checkov.cloudformation.checks.resource.base_resource_value_check import BaseResourceValueCheck
 from checkov.common.models.enums import CheckResult, CheckCategories
 from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
 
 
-class S3BlockPublicACLs(BaseResourceCheck):
+class S3BlockPublicACLs(BaseResourceValueCheck):
+    def get_inspected_key(self):
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/S3BlockPublicACLs.py:7]: @arielkru Same constructor first comment

File: checkov/cloudformation/checks/resource/aws/RedshiftClusterEncryption.py [hunk 11]
```diff
@@ -1,26 +1,17 @@
-from checkov.common.models.enums import CheckResult, CheckCategories
-from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
+from checkov.cloudformation.checks.resource.base_resource_value_check import BaseResourceValueCheck
+from checkov.common.models.enums import CheckCategories
 
 
-class RedshiftClusterEncryption(BaseResourceCheck):
+class RedshiftClusterEncryption(BaseResourceValueCheck):
+    def get_inspected_key(self):
```
Comment (nimrodkor) [inline checkov/cloudformation/checks/resource/aws/RedshiftClusterEncryption.py:6]: @arielkru Same constructor first comment

**Review by nimrodkor — COMMENTED** [review]
Awesome!!!

File: checkov/cloudformation/checks/resource/aws/cloudwatchLogGroupRetention.py (hunk 4 above)
Comment (schosterbarak) [reply checkov/cloudformation/checks/resource/aws/cloudwatchLogGroupRetention.py:18]: this is the terraform code:
not very different. @nimrodkor what do you mean?
```

class cloudwatchLogGroupRetention(BaseResourceValueCheck):
    def __init__(self):
        name = "Ensure cloudwatch log groups specify retention days"
        id = "CKV_AWS_66"
        supported_resource = ['aws_cloudwatch_log_group']
        categories = [CheckCategories.LOGGING]
        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resource)

    def get_inspected_key(self):
        return "retention_in_days"

    def get_expected_value(self):
        return ANY_VALUE
```

Comment (nimrodkor) [reply checkov/cloudformation/checks/resource/aws/cloudwatchLogGroupRetention.py:18]: I thought we had a limit there (less than X days). Guess I was wrong

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**